starlite/middleware/exceptions/debug_response.py

```diff
@@ -0,0 +1,186 @@
+        - we can not distinguish static methods from ordinary functions at the moment.
+
+    Returns:
+        A string containing full function name.
+    """
+
+    locals_dict = frame.frame.f_locals
+    # this piece assumes that the code uses standard names "self" and "cls"
+    # in instance and class methods
+    instance_or_cls = locals_dict.get("self") or locals_dict.get("cls")
+    classname = f"{get_name(instance_or_cls)}." if instance_or_cls else ""
+
+    return f"{classname}{frame.function}"
+
+
+def create_line_html(
+    line: str,
+    line_no: int,
+    frame_index: int,
+    idx: int,
+) -> str:
+    """Produce HTML representation of a line including real line number in the
+    source code.
+
+    Args:
+        line: A string representing the current line.
+        line_no: The line number associated with the executed line.
+        frame_index: Index of the executed line in the code context.
+        idx: Index of the current line in the code context.
+
+    Returns:
+        A string containing HTML representation of the given line
+    """
+    template = '<tr class="{line_class}"><td class="line_no">{line_no}</td><td class="code_line">{line}</td></tr>'
+    data = {
+        # line_no - frame_index produces actual line number of the very first line in the frame code context.
+        # so adding index (aka relative number) of a line in the code context we can calculate its actual number in the source file,
+        "line_no": line_no - frame_index + idx,
+        "line": escape(line).replace(" ", "&nbsp"),
+        "line_class": "executed-line" if idx == frame_index else "",
+    }
+    return template.format(**data)
+
+
+def create_frame_html(frame: "FrameInfo", collapsed: bool) -> str:
+    """Produce html representation of the given frame object including filename
+    containing source code and name of the function being executed.
+
+    Args:
+        frame: An instance of [FrameInfo](https://docs.python.org/3/library/inspect.html#inspect.FrameInfo).
+        collapsed: Flag controlling whether frame should be collapsed on the page load.
+
+    Returns:
+        A string containing HTML representation of the execution frame.
+    """
+    frame_tpl = (tpl_dir / "frame.html").read_text()
+
+    code_lines: List[str] = []
+    for idx, line in enumerate(frame.code_context or []):
+        code_lines.append(create_line_html(line, frame.lineno, frame.index or 0, idx))
+
+    data = {
+        "file": escape(frame.filename),
+        "line": frame.lineno,
+        "symbol_name": escape(get_symbol_name(frame)),
+        "code": "".join(code_lines),
+        "frame_class": "collapsed" if collapsed else "",
+    }
+    return frame_tpl.format(**data)
+
+
+def create_exception_html(exc: BaseException, line_limit: int) -> str:
+    """Produce html representation of the exception.
+
+    [frames](https://docs.python.org/3/reference/datamodel.html#frame-objects).
```

Comment:
```suggestion
    """Produce HTML representation of the exception
    [frames](https://docs.python.org/3/reference/datamodel.html#frame-objects).
```

Reply:
I think it is docformatter pre-commit lint that breaks the doc strings into multiple lines like this.

Reply:
Yes. You have to preemptively break the lines yourself. We should maybe see if we can fix that somehow.